src/karmen_backend/Dockerfile

```diff
@@ -1,41 +1,25 @@
-FROM python:3.7-alpine as builder
-RUN echo "http://dl-4.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
+FROM python:3.7-slim
 
-RUN mkdir /install
-WORKDIR /install
+# ensure www-data user
+# RUN set -x ; \
+#   groupadd --gid 82 --system www-data && \
+#   useradd --gid 82 --defaults --system -G www-data www-data && exit 0 || exit 1
 
-# build deps
-RUN apk --update add --no-cache gcc make postgresql-client postgresql-dev openssl-dev libffi-dev jq pcre-dev python3-dev build-base linux-headers libxml2-dev libxslt-dev
+WORKDIR /usr/src/app
 
-RUN pip install --upgrade pip
-RUN pip install uwsgi supervisor
+# build deps
+RUN set -e && apt-get update && apt-get install --yes git gcc make postgresql-client libffi-dev python3-dev libxml2-dev libxslt-dev libpq-dev \
+                      nginx gettext arp-scan postgresql-client bash dbus
 
+RUN set -e && pip install --upgrade pip pipenv uwsgi supervisor
 # Install from lockfile
+# there is a bug in current pipenv version when using --system flag
 COPY Pipfile* ./
-RUN jq -r '.default | to_entries[] | .key + .value.version' \
-    Pipfile.lock > requirements.txt && pip install -r requirements.txt
-
-
-FROM python:3.7-alpine as runner
```

Comment:
Removing build docker stage goes directly against future docker layer caching as the resulting images will be a lot bigger in size.

Reply:
It makes the resulting image slightly bigger. The caching is per `RUN` statement not per image. The caching should be the same.

Reply:
Yes, but the image to be downloaded to load the cached layers from will grow in size.

Reply:
Regarding cache size, it should be smaller if we remove the intermediate build server from dockerfile. The commands required to build the application will be cached in both cases. The difference is that in the later case (no intermediate build server) all layers will be shared with the final server (resulting in smaller cache but bigger final image). .... In theory. We need to verify it for sure.